NString/StringTemplateValueConverter.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text;
+using NString.Internal;
+
+namespace NString
+{
+    /// <summary>
+    /// Encapsulates a value conversion from an input value to an output value,
+    /// that will be used by <see cref="StringTemplate.Format(object, bool, IFormatProvider)"/>
+    /// instead of the original input value.
+    /// </summary>
+    public abstract class StringTemplateValueConverter<T> : IStringTemplateValueConverter
+    {
+        /// <summary>
+        /// Determines whether this instance can convert the specified object type.
+        /// </summary>
+        /// <param name="objectType">Type of the value to convert.</param>
+        /// <returns>
+        /// <c>true</c> if this instance can convert the specified object type; otherwise, <c>false</c>.
+        /// </returns>
+        public bool CanConvert(Type objectType)
```

Comment:
This should probably be virtual

Reply:
I'm not against it, but just so you know, `JsonConverter`'s equivalent method from Json.Net is not, and is actually explicitly sealed.

Reply:
Fair enough. I guess if someone really needs to change it, they can just implement the interface directly.